### DISCUSSION TITLE
Android installation issue

### DISCUSSION THREAD

**Discussion by adityarai004** [organicmaps/organicmaps#4494] [Q&A]
I am an android developer, chose to contribute here as numbers of issues are more here and i find the code and issues very understandable here.
I tried to set up the android code in my PC first it showed error bcoz of some secure.propeties that was solved with ./configure.sh
then after again an error shows up that was related to some sdk tool so solved it.
then i got some error in colors.xml and strings.xml they shows "..values-tr/colors.xml" got to know it is a issue due to symlink 
did the symlink many time even reinstalled git and CHECKED the BOX of symlink. constantly trying to find solution from last 6 hours found a solution but it is not solving my problem here the reference https://github.com/orgs/organicmaps/discussions/3095#discussioncomment-3436114

Now now I talked with some members they said to follow every instruction given in INSTALL.md then after literally deleted everything and restarted but no use.
Please Somebody do something about this issue feeling very demotivated rn. 
Expecting a solution asap

Windows 11 user
![image](https://user-images.githubusercontent.com/86659578/219101081-8e70247f-d117-421c-be9e-1bb2fbc55d5d.png)

**Comment by gpesquero** [comment]
Hello @adityarai004.

I’ve tried to get the Android development environment working on my Windows 10 system, and I've encountered the same problem as you have with the file `android\res\values-ar-rSY\colors.xml` not being symlinked to `..\values-tr\colors.xml`.

I've managed to fix the symlink issue enabling globally the symlinks by activating first the "Development Mode" in Windows Settings (first step mentioned here: https://github.com/orgs/organicmaps/discussions/3095#discussioncomment-3436114).

Can you please confirm that you have the developer mode activated in your Windows Settings? Because in your comment, you don't explicitely mention this step.

In Windows 10, "Development Mode" activation is in "Settings -> Update and Security -> For Developers", but in Windows 11 it's in "Settings -> Privacy & Security -> For Developers", as stated [here](https://pureinfotech.com/enable-developer-mode-windows-11/).

After enabling the Windows Development Mode, you have to clone again the OM repository following the installation instructions.

If you have already the Development Mode activated but the symlinks still don't work, then I honestly don't know what the problem is...

**Comment by adityarai004** [comment]
First of all do not forget to run "git config --global core.symlinks true"
in the cloned folder right click and then open git bash then run the
command I mentioned.
Then run ./configure.sh in the same folder of organic maps in git bash.
Then you will get the output like I have attached to this email.
If you too get output like this then
Open organic maps folder -> 3party -> boost here you will find
bootstrap.bat so just double click on it. A terminal will get opened up and
wait until it get CLOSED ITSELF. Do not force close it Maybe it will take
time.

When the terminal which popped up on double tapping bootstrap.bat file will
close it self then you will find that there is a new file in the same
folder where bootstrap.bat is called "b2" the extension would be .Exe just
run that b2.exe by double clicking it.

Lmk if you stuck again.
Regards,
Aditya

**Reply by Ashok-Kumar-dharanikota** [reply]
Hello @adityarai004 ,

I followed the steps you said. I didn't find any file named "bootstrap.bat". I found only "bootstrap.sh" and "bootstrap". can you please help me to setup android project.

**Reply by adityarai004** [reply]
Just double click on the "bootstrap" file, if a terminal opens let it be and have some patience till it get closed itself.
after it gets close there will be a file named "b2" you have to run it too by double clicking it.
remember you have to run bootstrap file till it gets completed only then you will find b2.

**Reply by biodranik** [reply]
All bat files and other commands should be run from the VS Developer Command Prompt, as it is described in instructions. Please also read what you see on the screen.

**Reply by Ashok-Kumar-dharanikota** [reply]
hi @adityarai004 ,

yesterday I tried, double clicked on the "bootstrap" file and the terminal opened within a few seconds it closed automatically. but I didn't find any file namely "b2". 

and @biodranik, I was running the commands in the android studio terminal and followed the same steps.
![Screenshot 2023-03-13 184553](https://user-images.githubusercontent.com/103619560/224713113-ad4b6c91-2cda-451a-99a6-64cd5962971d.png)
![Screenshot 2023-03-13 184618](https://user-images.githubusercontent.com/103619560/224713128-09ff1188-634d-4779-9c05-573534c47c18.png)

and i tried same steps in VS Developer CP. 
![Screenshot 2023-03-13 185022](https://user-images.githubusercontent.com/103619560/224713840-0a204532-f59a-4f32-a4d9-0eb0e2b3903c.png)

where did I get wrong?

**Reply by biodranik** [reply]
Interesting, looks like your "Developer command prompt" doesn't work for some reason. Are there any other Visual Studio shortcuts to launch other "dev command prompts"?

**Reply by Ashok-Kumar-dharanikota** [reply]
different dev command prompts. Here are a few of them:

1. To launch the Developer Command Prompt for Visual Studio: Press Ctrl + Q and type "Developer Command Prompt for VS" and press Enter.
2. To launch the Developer PowerShell for Visual Studio: Press Ctrl + Q and type "Developer PowerShell for VS" and press Enter.
3. To launch the x64 Native Tools Command Prompt: Press Ctrl + Q and type "x64 Native Tools Command Prompt" and press Enter.
4. To launch the x86 Native Tools Command Prompt: Press Ctrl + Q and type "x86 Native Tools Command Prompt" and press Enter.
5. To launch the ARM Cross Tools Command Prompt: Press Ctrl + Q and type "ARM Cross Tools Command Prompt" and press Enter.

is any of these command prompts help me?

**Comment by gpesquero** [comment]
Please just answer this simple question: Have you activated the *"Development Mode"* in your Windows 11 Settings?

As shown in the picture...

![enable-developer-mode-windows-11](https://user-images.githubusercontent.com/25888296/219855548-066ca967-dc8f-413c-8f8d-bca2d19c6876.png)

**Comment by adityarai004** [comment]
that is not necessary. My setup is successful with developer mode

**Comment by gpesquero** [comment]
Are you saying that your issue is solved and that you have a working Android development environment under Windows?
If so, then mark this discussion as answered....

**Reply by rtsisyk** [reply]
And please update [INSTALL.md](https://github.com/organicmaps/organicmaps/blob/master/docs/INSTALL.md) with actual instructions!

**Reply by gpesquero** [reply]
I've already launched [this](https://github.com/organicmaps/organicmaps/pull/4559) pull request to include the instructions to avoid symlinks problems...
